### DISCUSSION TITLE
HowTo Configure system's Bluetooth to tether with Smartphone to use the network of Cellular/mobile ISP?

### DISCUSSION THREAD

**Discussion by rajibando** [lxqt/lxqt#2613] [Q&A]
Please don't reply to this post presently. I have just begun editing. Over the next couple of days I will fully populate the post with the requisite input and data.
Info:  A standard laptop having debian-live-1 2.6.0-amd64-lxqt.iso installed
______________________________
connman and blueman (both preinstalled with the said Debian ISO) couldn't be used to activate accessing internet using smartphone's bluetooth tethering to use the mobile/cellular ISP's network.

In the said debian ISO, network-manager, nm-tray and network-manager-gnome don't come preinstalled. I installed them via synaptic.

` connman` was disabled with:
`sudo systemctl stop connman`
`sudo systemctl disable connman`

The Blueman > "Local Services" options window shows:
Network Settings
Services
Network Access Point (NAP)
NAP Settings
DHCP server type:
(Out of 3: udhcpd, dhcpd(3) and dnsmasq, 

I installed and selected dnsmasq.
IP Address is displayed as: 10.218.215.1
For PAN Support, What should I select from the two, NetworkManager
 and Blueman (dhclient)?

For DUN Support, what should I select from the two of NetworkManager and Blueman?

The screenshot provided below may please be perused:
![screen--13h50m29s17Sep24](https://github.com/user-attachments/assets/e0214de0-e2a7-41af-b7e4-32bb02cfca0d)

My nm-tray version is 0.5.0-1
My blueman version is 2.3.5-2+b1
They are as available for download from Debian repository with the reference from the Debian ISO. I wouldn't like to change them. If the system breaks, it would be beyond my ability to recover back the same set up again.

For my selection: dnsmasq, Blueman (dhclient) and Blueman respectively as said above, I still receive the **error code** on the **pop-up window** with the following title:
______________________________________
Failed To Apply Network Settings
Exception
g-dbus-error-quark: GDBus.Error:org.freedesktop.DBus.Error.Failed: Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/blueman/main/DbusService.py", line 156, in _handle_method_call
    ok(method(*args))
       ^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/blueman/plugins/mechanism/Network.py", line 52, in _enable_network
    nc.set_dhcp_handler(DHCPDHANDLERS[dhcp_handler])
  File "/usr/lib/python3/dist-packages/blueman/main/NetConf.py", line 385, in set_dhcp_handler
    self.dhcp_handler.do_apply()
  File "/usr/lib/python3/dist-packages/blueman/main/NetConf.py", line 113, in do_apply
    raise NetworkSetupError(f"dnsmasq failed to start: {error_msg}")
blueman.main.NetConf.NetworkSetupError: dnsmasq failed to start: dnsmasq: failed to bind DHCP server socket: Address already in use
 (0)
______________________________
It is curious to observe from the post of Bluey26 below that the **Network Access Point (NAP)** remains unticked, but the **dnsmasq** is opted for.  For PAN and DUN Support, NetworkManager is opted for, rather than Blueman.

My Debian ISO has `connman` installed, instead of `network-manager`, `network-manager-gnome` and `nm-tray`.  Though I have the three packages installed, I have the three presently kept them disabled to avoid their clash with `connman`.

A perusal of the [_other post on the issue_](https://github.com/lxqt/lxqt/discussions/2615) would be requested before a suggestion is posted.

**Comment by Bluey26** [comment]
Hello, i just wanted to say that this seems to work in here, without configuring NAP or a dhcpd server (see picture).

![blueman](https://github.com/user-attachments/assets/a14c338c-5217-4445-b699-0c93089ece4c)

I paired the device with the smartphone, then i enabled the "bluetooth tethering" option in the phone, which after a few seconds made available the internet connection in the laptop (internet from the smartphone).

You said that connman was the connection method you had installed. I do not understand much on how all those protocols work, but my first clue would be a problem with blueman itself.

I am using the following:
nm-tray (`nm-tray 0.5.0-2`)
blueman (`blueman 2.4.3`)

Check the versions you have of those packages, it could help to install newer ones.
